src/action/Skewer.cpp

```diff
@@ -13,114 +13,98 @@
 using ada::util::getRosParam;
 
 static const std::vector<std::string> optionPrompts{"(1) success", "(2) fail"};
-static const std::vector<std::string> actionPrompts{
-    "(1) skewer", "(3) tilt", "(5) angle"};
+static const std::vector<std::string> actionPrompts{"(1) skewer", "(3) tilt",
+                                                    "(5) angle"};
 
 namespace feeding {
 namespace action {
 
 //==============================================================================
-bool skewer(
-    const std::shared_ptr<Perception>& perception,
-    const std::string& foodName,
-    const Eigen::Isometry3d& plate,
-    const Eigen::Isometry3d& plateEndEffectorTransform,
-    FeedingDemo* feedingDemo)
-{
+bool skewer(const std::shared_ptr<Perception> &perception,
+            const std::string &foodName, const Eigen::Isometry3d &plate,
+            const Eigen::Isometry3d &plateEndEffectorTransform,
+            FeedingDemo *feedingDemo) {
   // Load necessary parameters from feedingDemo
-  const std::shared_ptr<::ada::Ada>& ada = feedingDemo->getAda();
-  const ros::NodeHandle* nodeHandle = feedingDemo->getNodeHandle().get();
-  const aikido::constraint::dart::CollisionFreePtr& collisionFree = feedingDemo->getCollisionConstraint();
-  const std::unordered_map<std::string, double>& foodSkeweringForces = feedingDemo->mFoodSkeweringForces;
+  const std::shared_ptr<::ada::Ada> &ada = feedingDemo->getAda();
+  const ros::NodeHandle *nodeHandle = feedingDemo->getNodeHandle().get();
+  const aikido::constraint::dart::CollisionFreePtr &collisionFree =
+      feedingDemo->getCollisionConstraint();
+  const std::unordered_map<std::string, double> &foodSkeweringForces =
+      feedingDemo->mFoodSkeweringForces;
   double heightAboveFood = feedingDemo->mFoodTSRParameters.at("height");
-  double rotationToleranceForFood = feedingDemo->mFoodTSRParameters.at("rotationTolerance");
+  double rotationToleranceForFood =
+      feedingDemo->mFoodTSRParameters.at("rotationTolerance");
   double moveOutofFoodLength = feedingDemo->mMoveOufOfFoodLength;
   std::chrono::milliseconds waitTimeForFood = feedingDemo->mWaitTimeForFood;
   // const Eigen::Vector6d& velocityLimits = feedingDemo.mVelocityLimits;
-  const std::shared_ptr<FTThresholdHelper>& ftThresholdHelper = feedingDemo->getFTThresholdHelper();
-  std::vector<std::string> rotationFreeFoodNames = feedingDemo->mRotationFreeFoodNames;
+  const std::shared_ptr<FTThresholdHelper> &ftThresholdHelper =
+      feedingDemo->getFTThresholdHelper();
+  std::vector<std::string> rotationFreeFoodNames =
+      feedingDemo->mRotationFreeFoodNames;
+
+  // TODO (egordon): add option to disable override in feeding_demo.yaml
```

Comment:
@amalnanavati This is where the YAML overrides the perception. This variable `actionOverride` was initially set to -1, and only used in the user study if the user explicitly selected an action.

Now, this variable is set based on the YAML (if it exists), and overrides perception.